diff -urN strace-4.3.org/stream.c strace-4.3/stream.c
--- strace-4.3.org/stream.c	Wed Apr  4 19:25:52 2001
+++ strace-4.3/stream.c	Wed Apr  4 20:22:05 2001
@@ -36,13 +36,8 @@
 #ifdef HAVE_SYS_POLL_H
 #include <sys/poll.h>
 #endif
+#include <stropts.h>
 
-#define RS_HIPRI 1
-struct strbuf {
-        int     maxlen;                 /* no. of bytes in buffer */
-        int     len;                    /* no. of bytes returned */
-        char    *buf;                   /* pointer to data */
-};
 #define MORECTL 1
 #define MOREDATA 2
 
