--- strace-4.5.14/defs.h.orig	2006-01-12 11:18:53.000000000 +0100
+++ strace-4.5.14/defs.h	2006-01-21 22:59:49.045368750 +0100
@@ -193,14 +193,15 @@
 
 #ifdef LINUXSPARC
 #include <linux/a.out.h>
-#include <asm/psr.h>
 #define PERSONALITY0_WORDSIZE 4
 #define PERSONALITY1_WORDSIZE 4
 #undef  SUPPORTED_PERSONALITIES
 #if defined(SPARC64)
+#include <asm/psrcompat.h>
 #define SUPPORTED_PERSONALITIES 3
 #define PERSONALITY2_WORDSIZE 8
 #else
+#include <asm/psr.h>
 #define SUPPORTED_PERSONALITIES 2
 #endif /* SPARC64 */
 #endif /* LINUXSPARC */
--- strace-4.5.15/mem.c.orig	2006-12-13 19:57:23.000000000 +0000
+++ strace-4.5.15/mem.c	2007-04-16 19:56:59.687032000 +0000
@@ -541,7 +541,7 @@
 	return 0;
 }
 
-#if defined(ALPHA) || defined(FREEBSD) || defined(IA64) || defined(SUNOS4) || defined(SVR4)
+#if defined(ALPHA) || defined(FREEBSD) || defined(IA64) || defined(SUNOS4) || defined(SVR4) || defined(SPARC) || defined(SPARC64)
 int
 sys_getpagesize(tcp)
 struct tcb *tcp;
