diff -Nur strace-4.5.1.orig/linux/alpha/syscallent.h strace-4.5.1/linux/alpha/syscallent.h
--- strace-4.5.1.orig/linux/alpha/syscallent.h	2003-07-09 11:47:59.000000000 +0200
+++ strace-4.5.1/linux/alpha/syscallent.h	2003-11-18 20:53:22.722489584 +0100
@@ -103,7 +103,7 @@
 	{ 5,	0,	printargs,		"osf_old_vadvise"	}, /* 72, not implemented */
 	{ 2,	0,	sys_munmap,		"munmap"		}, /* 73 */
 	{ 3,	0,	sys_mprotect,		"mprotect"		}, /* 74 */
-	{ 0,	0,	sys_madvise,		"madvise"		}, /* 75 */
+	{ 3,	0,	sys_madvise,		"madvise"		}, /* 75 */
 	{ 0,	0,	sys_vhangup,		"vhangup"		}, /* 76 */
 	{ 5,	0,	printargs,		"osf_kmodcall"		}, /* 77, not implemented */
 	{ 5,	0,	printargs,		"osf_mincore"		}, /* 78, not implemented */
@@ -399,12 +399,12 @@
 	{ 2,	0,	sys_capget,		"capget"		}, /* 368 */
 	{ 2,	0,	sys_capset,		"capset"		}, /* 369 */
 	{ 4,	TF,	sys_sendfile,		"sendfile"		}, /* 370 */
-	{ 3,	0,	printargs,		"setresgid"		}, /* 371 */
-	{ 3,	0,	printargs,		"getresgid"		}, /* 372 */
-	{ 4,	0,	printargs,		"dipc"			}, /* 373, not implemented */
-	{ 2,	TF,	printargs,		"pivot_root"		}, /* 374 */
-	{ 3,	0,	printargs,		"mincore"		}, /* 375 */
-	{ 4,	0,	printargs,		"pciconfig_iobase"	}, /* 376 */
+	{ 3,	0,	sys_setresgid,		"setresgid"		}, /* 371 */
+	{ 3,	0,	sys_getresgid,		"getresgid"		}, /* 372 */
+	{ 5,	0,	printargs,		"dipc"			}, /* 373, not implemented */
+	{ 2,	TF,	sys_pivotroot,		"pivot_root"		}, /* 374 */
+	{ 3,	0,	sys_mincore,		"mincore"		}, /* 375 */
+	{ 3,	0,	printargs,		"pciconfig_iobase"	}, /* 376 */
 	{ 4,	0,	sys_getdents64,		"getdents64"		}, /* 377 */
 	{ 0,	0,	printargs,		"gettid"		}, /* 378 */
 	{ 4,	0,	sys_readahead,		"readahead"		}, /* 379 */
--- strace-4.5.6/linux/sparc/syscallent.h.orig	2004-07-12 09:41:05.000000000 +0200
+++ strace-4.5.6/linux/sparc/syscallent.h	2004-07-26 08:21:58.496665928 +0200
@@ -80,7 +80,7 @@
 	{ 2,	0,	sys_getgroups,	"getgroups" },		/* 79 */
 	{ 2,	0,	sys_setgroups,	"setgroups" },		/* 80 */
 	{ 0,	0,	sys_getpgrp,	"getpgrp" },		/* 81 */
-	{ 2,	0,	sys_setgroups,	"setgroups32" },	/* 82 */
+	{ 2,	0,	sys_setgroups32,"setgroups32" },	/* 82 */
 	{ 3,	0,	sys_setitimer,	"setitimer" },		/* 83 */
 	{ 2,	0,	sys_ftruncate,	"ftruncate64" },	/* 84 */
 	{ 1,	TF,	sys_swapon,	"swapon" },		/* 85 */
@@ -113,7 +113,7 @@
 	{ 2,	TS,	sys_setresgid,	"setresgid32" },	/* 112 */
 	{ 5,	TN,	sys_recvmsg,	"recvmsg" },		/* 113 */
 	{ 5,	TN,	sys_sendmsg,	"sendmsg" },		/* 114 */
-	{ 2,	0,	sys_getgroups,	"getgroups32" },	/* 115 */
+	{ 2,	0,	sys_getgroups32,"getgroups32" },	/* 115 */
 	{ 2,	0,	sys_gettimeofday,"gettimeofday" },	/* 116 */
 	{ 2,	0,	sys_getrusage,	"getrusage" },		/* 117 */
 	{ 5,	TN,	sys_getsockopt,	"getsockopt" },		/* 118 */
diff -Nur strace-4.5.1.orig/linux/syscallent.h strace-4.5.1/linux/syscallent.h
--- strace-4.5.1.orig/linux/syscallent.h	2003-11-13 23:32:25.000000000 +0100
+++ strace-4.5.1/linux/syscallent.h	2003-11-18 21:43:16.371385776 +0100
@@ -248,7 +248,7 @@
 
 	{ 0,	0,	sys_getgid,		"getgid32"	}, /* 200 */
 	{ 0,	0,	sys_geteuid,		"geteuid32"	}, /* 201 */
-	{ 0,	0,	sys_geteuid,		"getegid32"	}, /* 202 */
+	{ 0,	0,	sys_getegid,		"getegid32"	}, /* 202 */
 	{ 2,	0,	sys_setreuid,		"setreuid32"	}, /* 203 */
 	{ 2,	0,	sys_setregid,		"setregid32"	}, /* 204 */
 	{ 2,	0,	sys_getgroups32,	"getgroups32"	}, /* 205 */
--- strace-4.5.1/linux/sparc/syscall.h.orig	2003-01-30 20:15:20.000000000 +0000
+++ strace-4.5.1/linux/sparc/syscall.h	2003-11-19 21:15:24.000000000 +0000
@@ -105,6 +105,7 @@
 int sys_rt_sigtimedwait(), sys_prctl(), sys_poll();
 int sys_sendfile(), sys_query_module(), sys_capget(), sys_capset();
 int sys_create_module(), sys_init_module();
+int sys_setgroups32(), sys_getgroups32();
 
 int	sys_umask();		/* XXX */
 
