diff -ur strace-4.2.orig/system.c strace-4.2/system.c
--- strace-4.2.orig/system.c	Thu Dec 23 15:20:15 1999
+++ strace-4.2/system.c	Sat Feb 12 19:46:51 2000
@@ -72,6 +72,8 @@
 #include <linux/sysctl.h>
 #endif
 
+#include <linux/version.h>
+
 static struct xlat mount_flags[] = {
 	{ MS_RDONLY,	"MS_RDONLY"	},
 	{ MS_NOSUID,	"MS_NOSUID"	},
@@ -1435,6 +1437,9 @@
 	{ KERN_PROF, "KERN_PROF" },
 	{ KERN_NODENAME, "KERN_NODENAME" },
 	{ KERN_DOMAINNAME, "KERN_DOMAINNAME" },
+#if LINUX_VERSION_CODE >= 131595
+	{ KERN_CAP_BSET, "KERN_CAP_BSET" },
+#endif
 #ifdef KERN_SECURELVL
 	{ KERN_SECURELVL, "KERN_SECURELVL" },
 #endif
