diff -urN strace-4.2.org/Makefile.in strace-4.2/Makefile.in
--- strace-4.2.org/Makefile.in	Sun Nov  5 17:56:54 2000
+++ strace-4.2/Makefile.in	Sun Nov  5 17:57:02 2000
@@ -39,8 +39,9 @@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 
-bindir = $(exec_prefix)/bin
-man1dir = $(prefix)/man/man1
+bindir = @bindir@
+mandir = @mandir@
+man1dir = $(mandir)/man1
 man1ext = .1
 
 SHELL = /bin/sh
diff -urN strace-4.2.org/system.c strace-4.2/system.c
--- strace-4.2.org/system.c	Sun Nov  5 17:56:54 2000
+++ strace-4.2/system.c	Sun Nov  5 17:57:02 2000
@@ -1435,6 +1435,9 @@
 	{ KERN_PROF, "KERN_PROF" },
 	{ KERN_NODENAME, "KERN_NODENAME" },
 	{ KERN_DOMAINNAME, "KERN_DOMAINNAME" },
+#ifdef KERN_CAP_BSET
+	{ KERN_CAP_BSET, "KERN_CAP_BSET" },
+#endif
 #ifdef KERN_SECURELVL
 	{ KERN_SECURELVL, "KERN_SECURELVL" },
 #endif
diff -urN strace-4.2.org/time.c strace-4.2/time.c
--- strace-4.2.org/time.c	Sun Nov  5 17:56:54 2000
+++ strace-4.2/time.c	Sun Nov  5 17:57:15 2000
@@ -33,7 +33,7 @@
 
 #ifdef LINUX
 #include <linux/version.h>
-#include <linux/timex.h>
+#include <sys/timex.h>
 #endif /* LINUX */
 
 void
