diff -urN strace-4.3.org/linux/sparc/syscall.h strace-4.3/linux/sparc/syscall.h
--- strace-4.3.org/linux/sparc/syscall.h	Wed Apr  4 19:25:52 2001
+++ strace-4.3/linux/sparc/syscall.h	Wed Apr  4 20:11:24 2001
@@ -82,7 +82,9 @@
 int	sys_fchdir(),sys_fchroot();
 int	sys_mkdir(),sys_rmdir(),sys_getdirentries(), sys_getdents(),sys_readdir();
 int	sys_creat(),sys_open(),sys_mknod(),sys_unlink(),sys_stat(),sys_fstat(),sys_lstat();
-int	sys_chown(),sys_fchown(),sys_chmod(),sys_fchmod(),sys_utimes();
+int	sys_stat64(), sys_lstat64(), sys_fstat64(), sys_truncate64(), sys_ftruncate64(), sys_madvise();
+int	sys_pivot_root();
+int	sys_lchown(),sys_fchown(),sys_chmod(),sys_fchmod(),sys_utimes();
 int	sys_link(),sys_symlink(),sys_readlink(),sys_rename();
 int	sys_lseek(),sys_truncate(),sys_ftruncate(),sys_access(),sys_fsync(),sys_sysctl();
 int	sys_statfs(),sys_fstatfs(),sys_msync();
diff -urN strace-4.3.org/linux/sparc/syscallent.h strace-4.3/linux/sparc/syscallent.h
--- strace-4.3.org/linux/sparc/syscallent.h	Wed Apr  4 19:25:52 2001
+++ strace-4.3/linux/sparc/syscallent.h	Wed Apr  4 20:10:43 2001
@@ -14,7 +14,7 @@
         { 3,    TF,     sys_chown,      "chown"},		/* 13 */
 	{ 3,	TF,	sys_mknod,	"mknod" },		/* 14 */
 	{ 2,	TF,	sys_chmod,	"chmod" },		/* 15 */
-	{ 3,	TF,	sys_chown,	"lchown" },		/* 16 */
+	{ 3,	TF,	sys_lchown,	"lchown" },		/* 16 */
 	{ 1,	0,	sys_brk,	"brk" },		/* 17 */
 	{ 2,	0,	printargs,	"perfctr" },		/* 18 */
 	{ 3,	0,	sys_lseek,	"lseek" },		/* 19 */
